src/pages/main/components/selected-dataset/build-flow/components/cards/presets-card/presets-card.store.ts

```diff
@@ -49,6 +50,14 @@ class PresetsCardStore {
     this.loadPresets()
     this.loadDTrees()
   }
+
+  public getSolutionsByRubric(rubric?: TExploreGenomeKeys) {
+    if (rubric) {
```

Comment:
return this.solutions.filter(s => rubric ? s === rubric : true)

Reply:
This would force to loop through the array even if the rubric is not presented. Not much efficient.